src/ucp/tag/rndv.c

```diff
@@ -847,80 +1001,183 @@ static ucs_status_t ucp_rndv_pipeline(ucp_request_t *sreq, ucp_rndv_rtr_hdr_t *r
     ucp_lane_index_t mem_type_rma_lane;
     ucp_ep_h mem_type_ep;
     ucp_mem_desc_t *mdesc;
-    ucp_request_t *frag_req;
+    ucp_request_t *freq;
+    ucp_request_t *fsreq;
     ucp_rsc_index_t md_index;
     ucs_status_t status;
     int i, num_frags;
-    size_t frag_size, length;
-    size_t offset;
-
-    ucs_assert(!UCP_MEM_IS_HOST(sreq->send.mem_type));
+    size_t max_frag_size, rndv_size, length;
+    size_t offset, rndv_base_offset;
 
     /* check if lane supports host memory, to stage sends through host memory */
     md_attr = ucp_ep_md_attr(sreq->send.ep, sreq->send.lane);
     if (!(md_attr->cap.reg_mem_types & UCS_BIT(UCS_MEMORY_TYPE_HOST))) {
         return UCS_ERR_UNSUPPORTED;
     }
 
-    /* check if there are rma_bw lanes (for get_zcopy) on memtype endpoint */
-    mem_type_ep       = worker->mem_type_ep[sreq->send.mem_type];
-    mem_type_rma_lane = ucp_ep_config(mem_type_ep)->key.rma_bw_lanes[0];
-    if (mem_type_rma_lane == UCP_NULL_LANE) {
-        return UCS_ERR_UNSUPPORTED;
+    rndv_size          = rndv_rtr_hdr->size;
+    max_frag_size      = context->config.ext.rndv_frag_size;
+    rndv_base_offset   = rndv_rtr_hdr->offset;
+    num_frags          = ucs_div_round_up(rndv_size, max_frag_size);
```

Comment:
seems too many spaces between filed names and `=`

Reply:
fixed